src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,295 @@
+
+        if (yield from bps.rd(self.oav_backlight.pos)) == 0:
+            yield from bps.abs_set(self.oav_backlight.pos, 1, wait=True)
+
+        """
+       TODO: currently can't find the backlight brightness
+        this will need to be set up after issue
+        https://github.com/DiamondLightSource/python-artemis/issues/317
+        is solved
+         brightnessToUse = self.oav_parameters._extract_dict_parameter(
+            context, "brightness"
+        )
+
+        blbrightness.moveTo(brightnessToUse)
+        """
+
+    def start_mxsc(self, input_plugin, min_callback_time, filename):
+        """
+        Sets PVs relevant to edge detection plugin.
+
+        Args:
+            input_plugin: link to the camera stream
+            min_callback_time: the value to set the minimum callback time to
+            filename: filename of the python script to detect edge waveforms from camera stream.
+        Returns: None
+        """
+        yield from bps.abs_set(self.oav.input_plugin_pv, input_plugin)
+
+        # Turns the area detector plugin on
+        yield from bps.abs_set(self.oav.enable_callbacks_pv, 1)
+
+        # Set the minimum time between updates of the plugin
+        yield from bps.abs_set(self.oav.min_callback_time_pv, min_callback_time)
+
+        # Stop the plugin from blocking the IOC and hogging all the CPU
+        yield from bps.abs_set(self.oav.blocking_callbacks_pv, 0)
+
+        # Set the python file to use for calculating the edge waveforms
+        yield from bps.abs_set(self.oav.py_filename_pv, filename, wait=True)
+        yield from bps.abs_set(self.oav.read_file_pv, 1)
+
+        # Image annotations
+        yield from bps.abs_set(self.oav.draw_tip_pv, True)
+        yield from bps.abs_set(self.oav.draw_edges_pv, True)
+
+        # Use the original image type for the edge output array
+        yield from bps.abs_set(
+            self.oav.output_array_pv, EdgeOutputArrayImageType.ORIGINAL
+        )
+
+    def _extract_beam_position(self):
+        """
+
+        Extracts the beam location in pixels `xCentre` `yCentre` extracted
+        from the file display.configuration. The beam location is manually
+        inputted by the beamline operator GDA by clicking where on screen a
+        scintillator ligths up.
+        """
+        with open(self.display_configuration_path, "r") as f:
+            file_lines = f.readlines()
+            for i in range(len(file_lines)):
+                if file_lines[i].startswith(
+                    "zoomLevel = " + str(self.oav_parameters.zoom)
+                ):
+                    crosshair_x_line = file_lines[i + 1]
+                    crosshair_y_line = file_lines[i + 2]
+                    break
+
+            if crosshair_x_line is None or crosshair_y_line is None:
+                pass  # TODO throw error
+
+            self.beam_centre_x = int(crosshair_x_line.split(" = ")[1])
+            self.beam_centre_y = int(crosshair_y_line.split(" = ")[1])
```

Comment:
Nit: This seems unrealted to the issue we're trying to solve? Can we do it in a separate PR please